### PR TITLE
TK-690: Handle memory bank renaming and changing head types.

### PR DESCRIPTION
**Pull request by Patronics**
This one isn't ready to merge yet, but I was wondering if I could get some tips on how to handle two inter-dependent settings, which I struggled with in commit 2924127e28?

Specifically, the channel name length and group name length must add up to the total display length, which is either 8 or 14 characters, depending on which head unit is equipped. I attempted to handle that logic, but haven't found a way to successfully update one setting in the UI when the other one's value changes. 

Similarly, even though I set the head unit type to be a volatile setting (as some buttons are only available on the full head unit), there's no change in the list of settings on the button_assignments settings tab after changing the head unit type.

### CODE REVIEW

**Comment by kk7ds** [comment]
> This one isn't ready to merge yet, but I was wondering if I could get some tips on how to handle two inter-dependent settings, which I struggled with in commit [2924127](https://github.com/kk7ds/chirp/commit/2924127e280798caed56bb71b0cb84b4fbd1bc8f)?
> 
> Specifically, the channel name length and group name length must add up to the total display length, which is either 8 or 14 characters, depending on which head unit is equipped. I attempted to handle that logic, but haven't found a way to successfully update one setting in the UI when the other one's value changes.

Yeah, volatile is the best we have here, so if you mark them both as such then you can make changes to the other and have the UI refreshed. Might be best to mark one as immutable (`set_mutable(False)`) and always make it calculated. So the user controls both by just setting one.

> Similarly, even though I set the head unit type to be a volatile setting (as some buttons are only available on the full head unit), there's no change in the list of settings on the button_assignments settings tab after changing the head unit type.

Yeah, you can't change the structure of the settings after we've built the UI. TBH, I'd just expose all of them with doc tips that the advanced ones (obviously) don't do anything if you don't have that head. However, I think you might also be able to mark them immutable/mutable based on the setting of the (volatile) head type. If the UI doesn't refresh the mutable-ness when it reloads after a volatile setting change we could probably fix that fairly easily.

The settings stuff is not what I would choose to design today, but some of the complexity comes from the fact that we support "live radios" as well, where settings (and all other) changes get flushed to the radio in real time.

**Comment by Patronics** [comment]
Thanks for the tips. Setting mutable to false for one of the settings makes the logic a bit simpler. Nothing seems to be reloading with volatile enabled, however. Pressing Command-R to reload the interface does update values accordingly. I'm going to see if I can spot any reasons the UI isn't refreshing as intended when volatile is active.

**Comment by Patronics** [comment]
Okay, did some digging into volatile's implementation, from what I can find, in common.py https://github.com/kk7ds/chirp/blob/605037deb68994bbb3114c2a35db7a7a7e4755a6/chirp/wxui/common.py#L474-L478 is successfully setting the needs_reload value, but in settings_edit.py https://github.com/kk7ds/chirp/blob/605037deb68994bbb3114c2a35db7a7a7e4755a6/chirp/wxui/settingsedit.py#L183-L187 when it tries to check the value from the propgrid, it gets the value as false, so the reload never happens. I'm not familiar with the details of the propertygrid implementation, but hopefully that gets you on the right track to solve it.

**Comment by kk7ds** [comment]
Okay yep, it was totally broken. Just happened to work for the driver I implemented it for because it was on the last page, which is silly of course. I imagine it was one of those "I'll fix this after I make sure it works" sort of things. So, new change at the front of your stack here. Please confirm it works for you and then I'll push it into the tree separately so you don't have to carry it.

**Comment by Patronics** [comment]
Okay, I just tested your fixes, it works great now, thanks for spending the time to figure it out!

**Comment by Patronics** [comment]
Okay, I've pushed some changes, adding a few more features (the power-on message and compander setting for each memory). Also fixed an issue I found with the default mode on the TK-690 model in particular. I think these changes should be ready to merge now :)

File: chirp/drivers/tk690.py [hunk 1]
```diff
@@ -1043,9 +1106,20 @@ def set_settings(self, settings):
...
-                    [value for (key, value) in BUTTON_FUNCTION_LIST
-                        if key == button.value.get_value()][0]
+                if groupKey == "button_assignments":
+                    self._memobj[groupKey][settingKey] = \
+                        [value for (key, value) in BUTTON_FUNCTION_LIST
+                            if key == button.value.get_value()][0]
+                elif groupKey == "basic_settings":
+                    if settingKey == "head_type":
+                        self._memobj.properties.rid.headtype = \
+                            button.value.__int__()
```
Comment (kk7ds) [inline chirp/drivers/tk690.py:1116]: Erm, this should be `int(button.value)` here (and below) right? You shouldn't be calling the dunder methods directly. Perhaps left over from debugging the lack of refresh thing?

File: chirp/drivers/tk690.py [hunk 2]
```diff
@@ -1008,33 +993,111 @@ def _update_bank_memmap(self):
...
+                              HEAD_TYPES,
+                              current_index=self._memobj.
+                              properties.rid.headtype))
+        rs.set_volatile(True)
+        basic_settings.append(rs)
+
+        ch_len_setting = RadioSettingValueInteger(0,
+                                                  14,
+                                                  self._memobj.settings
+                                                  .ch_name_length)
```
[the remark this answers is not in the document]
Comment (Patronics) [reply chirp/drivers/tk690.py:1061]: Yeah, I prefer your style too, I wasn't sure if tox would accept it, but since it does, I've adjusted that :)

**Pull request merged**